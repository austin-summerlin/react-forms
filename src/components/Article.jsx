import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, url }) => (
  <>
    <h1>title = {title}</h1>
    <p>author = {author}</p>
    <p>url = {url}</p>
  </>
);


Article.propTypes = {
  article: PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  })
};

export default Article;

