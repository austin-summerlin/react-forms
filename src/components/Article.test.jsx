/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('Atricle Component', () => {
  it('should render', () => {
    const article = {
      title: 'Test',
      author: 'Test Author',
      url: 'http://test.com',
    };

    const { asFragment } = render(<Article title={article.title} author={article.author} url={article.url} />);

    expect(asFragment).toMatchSnapshot();
  });
});
