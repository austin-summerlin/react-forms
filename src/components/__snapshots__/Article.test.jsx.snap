// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atricle Component should render 1`] = `[Function]`;
